src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupLoggingHelper.cs

```diff
@@ -16,6 +17,16 @@ namespace Microsoft.Build.BackEnd
     /// </summary>
     internal static class ItemGroupLoggingHelper
     {
+        /// <summary>
+        /// The default character limit for logging parameters. 10k is somewhat arbitrary, see https://github.com/microsoft/msbuild/issues/4907.
+        /// </summary>
+        internal static int parameterCharacterLimit = 10_000;
```

Comment:
```suggestion
        internal static int parameterCharacterLimit = 40_000;
```

? To just miss the LOH?

Reply:
Right